core/corehttp/gateway.go

```diff
@@ -25,11 +25,16 @@ func GatewayOption(writable bool, paths ...string) ServeOption {
 			return nil, err
 		}
 
+		api := coreapi.NewCoreAPI(n)
+		if cfg.Gateway.NoFetch {
+			api = api.(*coreapi.CoreAPI).Offline()
```

Comment:
What if we just made this a part of the API? We'll want a `ipfs --offline cat /...` eventually.

Reply:
+1, but this will be a slightly bigger change than I'm comfortable sneaking into this PR, opened an issue - https://github.com/ipfs/go-ipfs/issues/5654